Essentials/src/com/earth2me/essentials/commands/Commandpotion.java

```diff
@@ -39,28 +39,53 @@ protected void run(final Server server, final User user, final String commandLab
         }
         
         boolean holdingPotion = stack.getType() == Material.POTION;
-        if (!holdingPotion && ReflUtil.getNmsVersionObject().isHigherThanOrEqualTo(ReflUtil.V1_9_R1)) {
+        boolean holdingArrow = false;
+        boolean wasSpectral = false;
+
+        if (!holdingPotion && VersionUtil.v1_9_R01.isHigherThanOrEqualTo(VersionUtil.v1_9_R01)) {
             holdingPotion = stack.getType() == Material.SPLASH_POTION || stack.getType() == Material.LINGERING_POTION;
+            holdingArrow = stack.getType() == Material.ARROW;
+
+            if (!holdingArrow) {
+                if (stack.getType() == Material.TIPPED_ARROW || stack.getType() == Material.SPECTRAL_ARROW) {
+                    if (stack.getType() == Material.SPECTRAL_ARROW) {
+                        wasSpectral = true;
+                    }
+                    holdingArrow = true;
+                }
+            }
         }
-        if (holdingPotion) {
-            PotionMeta pmeta = (PotionMeta) stack.getItemMeta();
+
+        if (holdingPotion || holdingArrow) {
+            ItemStack potionStack = stack.clone();
+            if (holdingArrow) {
+                potionStack.setType(Material.TIPPED_ARROW);
+            }
+            PotionMeta pmeta = (PotionMeta) potionStack.getItemMeta();
             if (args.length > 0) {
                 if (args[0].equalsIgnoreCase("clear")) {
                     pmeta.clearCustomEffects();
-                    stack.setItemMeta(pmeta);
+                    potionStack.setItemMeta(pmeta);
                 } else if (args[0].equalsIgnoreCase("apply") && user.isAuthorized("essentials.potion.apply")) {
                     for (PotionEffect effect : pmeta.getCustomEffects()) {
                         effect.apply(user.getBase());
                     }
                 } else if (args.length < 3) {
                     throw new NotEnoughArgumentsException();
                 } else {
-                    final MetaItemStack mStack = new MetaItemStack(stack);
+                    final MetaItemStack mStack = new MetaItemStack(potionStack);
+                    if (wasSpectral) {
+                        mStack.addPotionMeta(user.getSource(), true, "effect:glowing", ess);
+                        mStack.addPotionMeta(user.getSource(), true, "duration:10", ess);
+                        mStack.addPotionMeta(user.getSource(), true, "power:1", ess);
+                    }
+
                     for (String arg : args) {
                         mStack.addPotionMeta(user.getSource(), true, arg, ess);
                     }
                     if (mStack.completePotion()) {
                         pmeta = (PotionMeta) mStack.getItemStack().getItemMeta();
+                        stack.setType(potionStack.getType());
```

Comment:
This entire diff needs to be cleaned up a lot: Use ternary operators and inline variable declaration (in the case of `holdingArrow` and `wasSpectral`